src/twisted/internet/process.py

```diff
@@ -163,7 +195,7 @@ def __init__(self, reactor, proc, name, fileno, forceReadHack=False):
             self.enableReadHack = False
         elif forceReadHack:
             self.enableReadHack = True
-        else:
+        elif not brokenCygwinPipeBehavior:
```

Comment:
my suggestion is to put this into a separate `if brokenCygwinPipeBehavior` and overwrite forceReadHack together with a comment